app/services/group_service.rb

```diff
@@ -41,6 +41,7 @@ def self.update(group:, params:, actor:)
 
   def self.archive(group:, actor:)
     actor.ability.authorize! :archive, group
+    SubscriptionService.new(group, actor).end_subscription!
```

Comment:
Hm, we definitely shouldn't be deleting either the Group or Subscription record in `archive!`; it looks like we're just setting the `archived_at` date on the group, so not sure why you wouldn't be able to find it again.